Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift

```diff
@@ -192,7 +192,9 @@ extension DefaultPaymentViewModel: PaymentViewControllerDelegate {
     // Check if security code is displayed and if so whether it is valid
     // This is business logic that wants Security code to be mandatory whenever its shown
     let isSecurityCodeRequired = paymentFormStyle?.securityCode != nil
-    if isSecurityCodeRequired && !cardValidator.isValid(cvv: cardDetails.cvv, for: cardScheme) { return }
+    if isSecurityCodeRequired {
+        guard let cvv = cardDetails.cvv, cardValidator.isValid(cvv: cvv, for: cardScheme) else { return }
+    }
```

Comment:
I am unsure how is this fixing it? The only way this may improve is if an empty CVV is valid option for the card scheme, which doesn't sound wrong?

Reply:
`securityCode` can be accept empty number string when card number is 19 digits. 
Issue is that `securityCode`  was before set to empty number string  when there's any error. 
Scenario: 
1. user enter in `card number` 19 digits `6799990100000000`
2. then user enter in `security code` value `0`
3. in our side, we validate `security code` then it's error 
4. then we return in the delegate `failure` then set `cardDetails.cvv` to be empty string ""
5. then we call `validateMandatoryInputProvided` 
6- then `isSecurityCodeRequired` is `true` `&&` `cardValidator.isValid(cvv: cardDetails.cvv, for: cardScheme)` is returning `true` then it's passing the `if condition`

Reply:
But all you've done with this PR is to make the default value `nil` instead of empty. As a rule of thumb we should avoid the ambiguity of nil.

Please try to add some CVV value and delete, then see if this code still validates correctly. Because your CVV will no longer be nil, it will be empty string, and exact same logic as previously will occur.